server/pbench/bin/pbench-server-prep-shim-001

```diff
@@ -0,0 +1,213 @@
+#! /bin/bash
+
+# set -x
+
+# This shim is used to prepare the tarballs that a version 001 client
+# submits for further processing. It copies the tarballs and their MD5
+# sums (plus an optional prefix file) to the archive (after checking)
+# and sets the state links, so that the dispatch script will pick them
+# up and get the ball rolling. IOW, it does impedance matching between
+# version 001 clients and the server scripts.
+
+# There is an enormous amount of code duplication between this script
+# and the version 002 script. TBD: refactor (but maybe wait and do it
+# for the inotify versions - the crontab version has a very limited life
+# span at this point).
+
+###########################################################################
+# load common things
+opts=$SHELLOPTS
+case $opts in
+    *xtrace*)
+        dir=$(dirname $(which $0))
+        PROG=$(basename $(which $0))
+        ;;
+    *)
+        dir=$(dirname $0)
+        PROG=$(basename $0)
+        ;;
+esac
+
+. $dir/pbench-base.sh
+###########################################################################
+
+archive=$(getconf.py archive-dir pbench-server)
+qdir=$(getconf.py quarantine-dir pbench-server)
+quarantine=${qdir}/md5-001
+duplicates=${qdir}/duplicates-001
+
+# The following directory holds tarballs that are quarantined because
+# of operational errors on the server. They should be retried after
+# the problem is fixed: basically, move them back into the reception
+# area for 001 agents and wait.
+errors=${qdir}/errors-001
+
+errlog=$LOGSDIR/$PROG/$PROG.error
+if [[ "$archive" == "" ]] ;then
+    echo "Failed: \"getconf.py archive-dir pbench-server" >> $errlog
+    exit 1
+fi
+
+if [[ "$quarantine" == "" ]] ;then
+    echo "Failed: \"getconf.py quarantine-dir pbench-server" $errlog
+    exit 1
+fi
+
+if [[ "$duplicates" == "" ]] ;then
+    echo "Failed: \"getconf.py duplicates-dir pbench-server" $errlog
+    exit 1
+fi
+
+log_init $(basename $0) $LOGSDIR/$(basename $0)
+
+tmp=$TMP/$PROG.$$
+
+trap 'rm -rf $tmp' EXIT INT QUIT
+
+mkdir -p $tmp
+list=$tmp/list.tb
+status=$tmp/status
+> $status
+# we are explicitly handling version-001 data
+results_dir=$(getconf.py receive-dir-001 pbench-server)
+
+# check for results that are ready for processing: version 001
+# agents create a link in the TODO subdir (which is created if
+# necessary) after they finish with MD5 checking, so that's what
+# we look for.
+find ${results_dir} -maxdepth 3 -type l -name '*.tar.xz' > $list
+
+typeset -i ntbs=0
+typeset -i nerrs=0
+typeset -i nquarantined=0
+typeset -i ndups=0
+
+echo $TS
```

Comment:
Given the `find` command can take a long time, should we move this `echo` of the start timestamp up before we issue the `find`?

Reply:
TS is set by pbench-base.sh once and for all. The only thing that moving it will accomplish will be to add it to the log file earlier. I can do that but unless somebody is actively looking at the log file, it won't make any difference.